projects/storefrontlib/src/lib/cms/components/dynamic-slot/dynamic-slot.component.html

```diff
@@ -4,7 +4,9 @@
       let component of ((currentSlot$ | async)?.components | slice: 0:limit)
     "
   >
-    <ng-container *cxOutlet="component.typeCode">
+    <ng-container
+      *cxOutlet="getComponentType(component.typeCode, component.uid)"
```

Comment:
What about pass in the component as an argument, make it type save (introduce `cmsComponentData` next to `ContentSlotData`)? It simplifies the view a bit.